inference/models/sam2/segment_anything2.py

```diff
@@ -0,0 +1,282 @@
+import base64
+import hashlib
+import os
+from io import BytesIO
+from time import perf_counter
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+import rasterio.features
+import torch
+from sam2.build_sam import build_sam2
+from sam2.sam2_image_predictor import SAM2ImagePredictor
+from shapely.geometry import Polygon as ShapelyPolygon
+
+from inference.core.entities.requests.inference import InferenceRequestImage
+from inference.core.entities.requests.sam2 import (
+    Sam2EmbeddingRequest,
+    Sam2InferenceRequest,
+    Sam2SegmentationRequest,
+)
+from inference.core.entities.responses.sam2 import (
+    Sam2EmbeddingResponse,
+    Sam2SegmentationResponse,
+)
+from inference.core.env import SAM2_VERSION_ID, SAM_MAX_EMBEDDING_CACHE_SIZE
+from inference.core.models.roboflow import RoboflowCoreModel
+from inference.core.utils.image_utils import load_image_rgb
+from inference.core.utils.postprocess import masks2poly
+
+
+class SegmentAnything2(RoboflowCoreModel):
+    """SegmentAnything class for handling segmentation tasks.
+
+    Attributes:
+        sam: The segmentation model.
+        predictor: The predictor for the segmentation model.
+        ort_session: ONNX runtime inference session.
+        embedding_cache: Cache for embeddings.
+        image_size_cache: Cache for image sizes.
+        embedding_cache_keys: Keys for the embedding cache.
+        low_res_logits_cache: Cache for low resolution logits.
+        segmentation_cache_keys: Keys for the segmentation cache.
+    """
+
+    def __init__(self, *args, model_id: str = f"sam2/{SAM2_VERSION_ID}", **kwargs):
+        """Initializes the SegmentAnything.
+
+        Args:
+            *args: Variable length argument list.
+            **kwargs: Arbitrary keyword arguments.
+        """
+        super().__init__(*args, model_id=model_id, **kwargs)
+        checkpoint = self.cache_file("weights.pt")
+        model_cfg = {
+            "hiera_large": "sam2_hiera_l.yaml",
```

Comment:
Did we put these in our weights bucket?

Reply:
Yes

Reply:
All 4 weights are tested in the nvidia-t4 workflow, a passing run is here: https://github.com/roboflow/inference/actions/runs/10210859531